src/DynamoCoreWpf/ViewModels/Core/InPortViewModel.cs

```diff
@@ -276,8 +288,16 @@ private void RefreshPortDefaultValueMarkerVisible()
         /// </summary>
         protected override void RefreshPortColors()
         {
-            //Is in function state
-            if (node.NodeModel.IsPartiallyApplied)
+            //Boolean that defines if the node has a default value from the cache value
+            isReturningDefaultValue = !string.IsNullOrEmpty(node.NodeModel.CachedValue?.StringData);
+            
+            //This variable checks if the node is a function class
+            bool isFunctionNode = node.NodeModel.CachedValue != null &&
```

Comment:
I think - there are better options - this is too fragile and could break. Let me look.